### ISSUE TITLE
include tags not working with ansible 1.8.1

### ISSUE THREAD

**Issue by passie** [ansible/ansible#9862]
##### Issue Type:

Bug Report
##### Ansible Version:

1.8.1-1.el6
##### Environment:

CentOS release 6.6 (Final)
##### Summary:

Tags from includes are being ignored
Tags from within a playbook are working just fine.

After a downgrade to Ansible 1.7.2-2.el6 this is working
##### Steps To Reproduce:

include: someotherplaybook.yml tags=test_tag 
ansible-playbook someplaybook.yml -e "target=somehost" -t test_tag 
##### Expected Results:

Tags should not be ignored
##### Actual Results:

ERROR: tag(s) not found in playbook: test_tag.  possible values:

**Comment by hkariti** [comment]
Still in 1.8.2 and HEAD

**Comment by natefoo** [comment]
I can confirm this on 1.8.2.

**Comment by kportertx** [comment]
Can confirm in 1.6.0, 1.7.2, 1.8.0, and 1.8.2.

Current behavior conflicts with docs: http://docs.ansible.com/playbooks_tags.html

The main playbook includes other playbooks which include roles

Below is an example of the structure I have
**site.yml**

```
- include: deploy-component-playbook.yml
  tags:
    - deploy-component
```

**deploy-component-playbook.yml**

```
- name: Deploy Component
  hosts: all
  roles:
    - role: deploy-subcomponent
```

**roles/deploy-subcomponent/tasks/main.yml**

```
- name: task name
  sudo: yes
  yum:
    name: ntpd
    state: present
```

**Comment by kristous** [comment]
Problem still in 1.8.4

**Comment by dkoppstein** [comment]
Still open in 1.9.0.1. Applies to roles as well.

**Comment by kportertx** [comment]
@dkoppstein is this issue applying to roles a regression in 1.9.0.1? I'm on 1.6.0 and can use tags on roles.

**Comment by dkoppstein** [comment]
Sorry, I had a stupid bug =) I confirm that tags work on roles in 1.9.0.1. Don't know about includes though.

**Comment by passie** [comment]
Any news on when this will be included in the main version ?

**Comment by bcoca** [comment]
so tags should work in roles and includes in 1.9.1 (currently rc 1).

**Comment by passie** [comment]
Looks like it isn't working in 1.9.1.1.el6

**Comment by bindlestifff** [comment]
Tagged include statements are still not working as described in the documentation in 1.9.1.  For those of us who depend on this behavior it has us stuck way back on 1.7.2 until it is fixed.

**Comment by AgentO3** [comment]
Yep, waiting for a fix before upgrade past 1.7.2 :( I got a production ansible repo that makes regular use of tagged include statements.

**Comment by bcoca** [comment]
So my tests show tags working on task includes and roles, can i get an example of this not working?

**Comment by bindlestifff** [comment]
#### testtag.yml

```
---
- include: playbooks/start-or-stop-service.yml tags=deployment
```

#### playbook/start-or-stop-service.yml

```
---
- name: Start or stop the service
  tasks:
    - name: Start service
      shell: /etc/init.d/myservice restart
```

```
ansible-playbook -t deployment -u user -k -i inventory/environment  playbooks/tagtest.yml 
```

That general setup works in 1.7.2 but not in any version we've tested since then including 1.9.1.  In 1.7.2 the "Start service" task will run but in subsequent versions there is no indication that the include ever happens.  I haven't retested with roles in the latest version but I'm certain it's not happening for includes.  Adding the 'deployment' tag in the included playbook tasks will enable them to run but according to docs and prior behavior that shouldn't be necessary.

**Comment by bcoca** [comment]
ah, my tests all are task includes or roles, not play includes, I'm not sure that those were ever meant to be a feature

**Comment by bcoca** [comment]
FYI, i consulted with core and yes, this is something that we need to fix

**Comment by AgentO3** [comment]
:+1:

**Comment by kportertx** [comment]
@bcoca The patch I provided above handles this case, been using it since I provided it.

**Comment by jinayvora1990** [comment]
@kportertx, @bcoca - Whats the status of this patch? When will it get merged to master?

**Comment by bcoca** [comment]
this is now working in the v2 engine and will probably be released with that version

**Comment by rgevaert** [comment]
I've also been bitten by this.  So would be nice to know when it could get into 1.9.  Thanks

**Comment by bcoca** [comment]
task includes are in 1.9.1 and above, for play includes this will be in 2.0

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by AshwinSridhar** [comment]
Is this fixed ? I don't think tags work along include in 1.9.4. When I use tags with include, all the tasks are executed and not the specific tagged task

**Comment by r66wen** [comment]
still not fixed in ansible 2.3.0.0

**Comment by neuroit** [comment]
I have the same issue in 2.3.2.0; In some playbooks it works, but in some it doesn't.
